src/BotiumBotmockExporter.ts

```diff
@@ -0,0 +1,443 @@
+import "dotenv/config";
+import {BaseExporter, Resources, FileWriter, Kind, ProjectReference} from "@botmock/export";
+import {Botmock, DataTransformation, KeyDependent} from "@botmock/export/src/types/index";
+import {BotDriver} from "botium-core"
+import {sanitize} from "sanitize-filename-ts";
+
+const SCRIPTING_FORMAT_TXT = 'SCRIPTING_FORMAT_TXT'
+const SCRIPTING_TYPE_CONVO = 'SCRIPTING_TYPE_CONVO'
+const SCRIPTING_TYPE_UTTERANCES = 'SCRIPTING_TYPE_UTTERANCES'
+
+import {Botium} from "./types";
+
+interface Log {
+    // node or connection
+    type: string,
+
+    connectionBranchingReason?: Botmock.Message['next_message_ids'][0],
+    connectionBranchCount?: number,
+    connectionBranchIndex?: number,
+    // the connection itself is extracted
+    connectionTerminatedReason?: string
+
+    // just for node type
+    nodeName?: string,
+    nodeMessageId?: string,
+    nodeMessageType?: string,
+    // the node itself is extracted
+    nodeTerminatedReason?: string
+}
+
+interface ResultEntry {
+    botiumConvoSteps: Botium.ConvoStep[],
+    processedMessageId: string[],
+    logs: Log[],
+}
+
+interface Utterance {
+    name: string,
+    utterances: string[]
+}
+
+export class BotiumBotmockExporter extends BaseExporter {
+    #CONVO_POSTFIX = ".convo.txt";
+    #UTTERANCES_POSTFIX = ".utterances.txt";
+
+    #getFallbackSafe = (obj: any, key: string) => {
+        if (!obj) {
+            return null;
+        }
+
+        if (obj[key]) {
+            return (obj[key]);
+        }
+
+        const keys = obj.keys();
+        if (keys.length) {
+            return null;
+        }
+
+        return obj[key[0]];
+    };
+
+    #getMessages = (payload: any, componentType: string): Botmock.Block[] => {
+        return (this.#getFallbackSafe(this.#getFallbackSafe(payload, 'en'), 'generic')?.blocks || []).filter((block: Botmock.Block) => block.component_type === componentType)
+    }
+
+    #intentToUtteranceRef = (intent: string): string => {
+        return `UTT_${intent.split(' ').join('_').toUpperCase()}`
+    }
+
+    /**
+     * Replaces botmock variables in text with botium variables
+     * @param text Text in block
+     */
+    #replaceVariableCharacterInTextDeprecated = (text: string): string => {
+        const alphanumericRegexp = new RegExp(/\%([a-zA-Z0-9_]+)\%/g);
```

Comment:
The \ is not needed before %:
```suggestion
        const alphanumericRegexp = new RegExp(/%([a-zA-Z0-9_]+)%/g);
```